src/control/server/server_utils.go

```diff
@@ -179,26 +175,27 @@ func getEngineNUMANodes(log logging.Logger, engineCfgs []*engine.Config) []strin
 	return nodes
 }
 
+// Prepare bdev storage. Assumes validation has already been performed on server config. Hugepages
+// are required for both emulated (AIO devices) and real NVMe bdevs. VFIO and IOMMU are not
+// required for emulated file launchedNVMe.
 func prepBdevStorage(srv *server, iommuEnabled bool) error {
 	defer srv.logDuration(track("time to prepare bdev storage"))
 
-	hasBdevs := cfgHasBdevs(srv.cfg)
+	if srv.cfg.DisableHugepages {
```

Comment:
If there are bdevs defined, shouldn't we be returning an error here?

Reply:
That validation will already have been performed on the config, don't think we need to validate here as well. I can check here as well if you feel strongly but that case (disable_hugepages set when bdevs exist) is tested in TestServer_prepBdevStorage which runs config validation before calling prepBdevStorage()